app/helpers/course_year_helper.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module CourseYearHelper
+  def get_user_lessons_and_progresses(ect_profile, course_lessons = nil)
+    course_lessons ||= CourseLesson.where(course_module: course_modules)
```

Comment:
That is a funny line. I imagine `course_modules` will not be recognised inside `CourseModule` :)
I would get rid of it, and pass course lessons to this method in both cases.